### PR TITLE
Refs [TASK][37121] Data Player Details Screen

### PR DESCRIPTION
**Pull request by zGararz**
## Related Tickets
- [#Task 37121: Data Player Details Screen](https://edu-redmine.sun-asterisk.vn/issues/37121)
## WHAT
- 
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/loltournaments/ui/player/PlayerDetailsFragment.kt [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
-import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
 import com.sunasterisk.loltournaments.R
+import com.sunasterisk.loltournaments.base.BaseFragment
+import com.sunasterisk.loltournaments.databinding.FragmentPlayerDetailsBinding
+import org.koin.androidx.viewmodel.ext.android.viewModel
 
-class PlayerDetailsFragment: Fragment(R.layout.fragment_player_details) {
+class PlayerDetailsFragment : BaseFragment<FragmentPlayerDetailsBinding>() {
+    override val layoutResource = R.layout.fragment_player_details
```
Comment (phucnv-2558) [inline app/src/main/java/com/sunasterisk/loltournaments/ui/player/PlayerDetailsFragment.kt]: auto dùng get() nhé

File: app/src/main/java/com/sunasterisk/loltournaments/data/source/remote/ApiService.kt [hunk 2]
```diff
@@ -23,7 +23,7 @@ interface ApiService {
         @Query(QUERY_FILTER_ID) tournamentId: Int
     ): List<Tournament>
 
-    @GET(PATH_MATCH)
+    @GET("$PATH_MATCH?$QUERY_SEARCH_PER_PAGE_100")
     suspend fun getMatchByTournamentId(
         @Query(QUERY_FILTER_TOURNAMENT_ID) matchId: Int
     ): List<Match>
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/loltournaments/data/source/remote/ApiService.kt]: Sao lại thêm query như thế em ? 
```suggestion
    @GET(PATH_MATCH)
    suspend fun getMatchByTournamentId(
        @Query(QUERY_FILTER_TOURNAMENT_ID) matchId: Int, 
        @Query("per_page") perPage: Int = 100
    ): List<Match>
```

File: app/src/main/java/com/sunasterisk/loltournaments/ui/player/PlayerDetailsFragment.kt [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+    override fun initData() {
+        binding.apply {
+            lifecycleOwner = viewLifecycleOwner
+            playerViewModel = viewModel
+        }
+
+        arguments?.let {
+            val arg = PlayerDetailsFragmentArgs.fromBundle(it)
+            viewModel.setPlayer(arg.player)
+        }
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/loltournaments/ui/player/PlayerDetailsFragment.kt]: sao không dùng safe args em ? mà lại phải dùng arguments chay như này

File: app/src/main/java/com/sunasterisk/loltournaments/data/source/remote/LolRemoteDataSource.kt [hunk 4]
```diff
@@ -8,7 +8,7 @@ class LolRemoteDataSource(
     override suspend fun getLeagues(leagueIds: String, sortValue: String) = apiService.getLeagues(leagueIds, sortValue)
     override suspend fun getSeries(time: String, leagueId: Int) = apiService.getSeries(time, leagueId)
     override suspend fun getTournamentById(tournamentId: Int) = apiService.getTournamentById(tournamentId)
-    override suspend fun getMatchByTournamentId(matchId: Int) = apiService.getMatchByTournamentId(matchId)
+    override suspend fun getMatchByTournamentId(matchId: Int) = apiService.getMatchByTournamentId(matchId = matchId)
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/loltournaments/data/source/remote/LolRemoteDataSource.kt]: viết như anh suggest, em sẽ k phải sửa code như này

**Pull request merged**